src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -986,6 +990,26 @@ impl LinkCollector<'_, '_> {
                 }
             }
 
+            if path_str.contains(['<', '>'].as_slice()) {
+                stripped_path_string = match strip_generic_parameters_from_path(path_str) {
+                    Ok(path) => path,
+                    Err(err_kind) => {
+                        debug!("link has malformed generics: {}", path_str);
+                        resolution_failure(
+                            self,
+                            &item,
+                            path_str,
+                            disambiguator,
+                            dox,
+                            link_range,
+                            smallvec![err_kind],
+                        );
+                        return;
```

Comment:
At some point I should probably change `resolve_link` to return an error so you can use `?` for it. No need to do that here though.